### DISCUSSION TITLE
Configure Grafana data source with 2 headers

### DISCUSSION THREAD

**Discussion by velu4689** [grafana/mimir#5076] [Development]
Hi Team,

Currently we are remote writing the prometheus data into Mimir cluster under 2 different headers (X-Scope-OrgID). 

So, are we good to add both headers into a single data source in Grafana or we need to configure them as separate data sources ?

Can you pls highlight the advantages & disadvantages ?

Thanks

**Comment by pstibrany** [comment]
Hi, correct answer really depends on your use case. You could query both tenants at the same time by enabling `-tenant-federation.enabled` in Mimir, and then setting `X-Scope-OrgID` header to `tenant1|tenant2`.

**Comment by velu4689** [comment]
@pstibrany: Thanks. in this case, I would like to add both headers ( tenant1 & tenant2) in the grafana data source. is that feasible ?

**Reply by pstibrany** [reply]
You can only add one header, but you can put multiple tenants into it, separated by `|`, as mentioned in my previous reply. To make this work you will need to run Mimir read components with `-tenant-federation.enabled` option.

**Comment by velu4689** [comment]
@pstibrany : Thanks. it worked

**Reply by Shindek77** [reply]
@pstibrany and @velu4689 I Have tried by enabling -tenant-federation.enabled as true in mimir values.yaml file as given below:
tenant_federation:
  enabled: true

compactor:
  replicas: 1
  service:
    annotations: {}
    labels: {}

After that I have added X-Scope-OrgID header to tenant1|tenant2 in grafana datasource of mimir. but I am not able to see metrics from both tenant.

@velu4689 Could you please provide me your configurations
